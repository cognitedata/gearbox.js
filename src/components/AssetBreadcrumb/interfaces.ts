import { Asset, CogniteInternalId } from '@cognite/sdk';

export type FetchAssetCall = (assetId: CogniteInternalId) => Promise<Asset>;

export interface AssetBreadcrumbProps {
  /**
   * Asset ID
   */
  assetId: number;
  /**
   * Maximal number of assets to be displayed.
   * If length of the asset chain bigger than < maxLength > value,
   * asset chain will be shrunk to root element plus < maxLength > - 1 number
   * of last elements in a chain
   */
  maxLength?: number;
  /**
   * Function, that can be used for custom rendering of asset in a breadcrumb
   */
  renderItem?: (asset: Asset, depth: number) => JSX.Element;
  /**
   * Function that can be used to replace embedded asset fetching
   */
  retrieveAsset?: FetchAssetCall;
  /**
   * Callback which is executed on a click action on asset in a breadcrumb.
   * It's only available in case of default asset rendering in a breadcrumb
   */
  onBreadcrumbClick?: (asset: Asset, depth: number) => void;
}
