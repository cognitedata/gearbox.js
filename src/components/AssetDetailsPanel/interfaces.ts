import { CSSProperties } from 'react';
import { WithAssetDataProps, WithAssetProps } from '../../hoc';
import { MetaDescriptionListProps } from '../DescriptionList';

export interface AssetDetailsPanelStylesProps {
  /**
   * Object that defines inline CSS style for container of the table
   */
  styles?: CSSProperties;
}

export type AssetDetailsPanelPureProps = WithAssetDataProps &
  MetaDescriptionListProps &
  AssetDetailsPanelStylesProps;

export type AssetDetailsPanelProps = WithAssetProps &
  MetaDescriptionListProps &
  AssetDetailsPanelStylesProps;
