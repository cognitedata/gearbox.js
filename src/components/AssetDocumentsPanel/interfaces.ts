import { FilesMetadata } from '@cognite/sdk';
import { CollapseProps } from 'antd/lib/collapse';
import { CSSProperties, ReactNode } from 'react';
import { WithAssetFilesDataProps, WithAssetFilesProps } from '../../hoc';
import { AnyIfEmpty } from '../../interfaces';

export type Document = FilesMetadata;

export type DocumentRenderer = (
  document: Document,
  i: number,
  documents: Document[]
) => ReactNode;

export type OnDocumentClick = (
  document: Document,
  category: string,
  description: string
) => void;

export interface JsonDocTypes {
  [s: string]: string;
}

export interface DocumentType {
  key: string;
  description: string;
}

export interface DocumentsByCategory {
  [s: string]: {
    description: string;
    documents: Document[];
  };
}

export interface Priority {
  [s: string]: number;
}

export interface DocumentsPanelStylesProps {
  /**
   * Object that defines inline CSS styles for inner elements of the component
   */
  styles?: AssetDocumentsPanelStyles;
}

export interface DocumentsPanelThemeProps {
  theme?: AnyIfEmpty<{}>;
}

export type DocumentTableProps = MetaDocProps &
  WithAssetFilesDataProps &
  DocumentsPanelStylesProps &
  DocumentsPanelThemeProps;

export interface AssetDocumentsPanelStyles {
  wrapper?: CSSProperties;
  fileContainer?: CSSProperties;
  fileLink?: CSSProperties;
  fileTitle?: CSSProperties;
}

export interface MetaDocProps {
  /**
   * Callback function triggered when user clicks on a file (document)
   */
  handleDocumentClick?: OnDocumentClick;
  /**
   * Object with props to be passed to atn design Collapse component
   */
  collapseProps?: CollapseProps;
  /**
   * List of categories codes to be shown on the top of the list.
   * Categories "P&ID" and "Logic Diagrams" are prioritized by default.
   */
  categoryPriorityList?: string[];
  /**
   * Name for the category that includes all files with undefined category
   */
  unknownCategoryName?: string;
  /**
   * Key to get document title from `metadata` object
   */
  documentTitleField?: string;
  /**
   * The `metadata` field used to group documents into types. If this is
   * not specified, then `doc_type` field will be attempted. If no
   * `doc_type` field is present in the metadata, then the filename will
   * be attempted to be parsed using the NORSOK standard. If this fails, then
   * the document will be in the "unknown" group.
   */
  documentTypeField?: string;
  /**
   * Object map used to show custom category names
   */
  docTypes?: JsonDocTypes;
  /**
   * Text to be shown if no documents have been found for the asset
   */
  noDocumentsSign?: string;
  /**
   * Custom render function to modify documents appearing
   */
  documentRenderer?: DocumentRenderer;
  /**
   * Function to sort categories after theirs prioritization
   */
  customCategorySort?: (a: string, b: string) => number;
}

export type AssetDocumentsPanelProps = WithAssetFilesProps &
  MetaDocProps &
  DocumentsPanelStylesProps;
