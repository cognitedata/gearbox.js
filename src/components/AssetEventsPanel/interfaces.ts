import { CSSProperties } from 'react';
import { WithAssetEventsDataProps, WithAssetEventsProps } from '../../hoc';
import { AnyIfEmpty } from '../../interfaces';

export interface TableColumnType {
  title: string;
  dataIndex: string;
  key?: string;
}

export interface TableDesignType {
  /**
   * Configure table pagination appearance
   */
  pagination?: {
    pageSize?: number;
    position?: 'top' | 'bottom' | 'both';
    showSizeChanger?: boolean;
  };
  /**
   * Configure scroll appearance
   */
  scroll?: {
    y?: string;
    x?: string;
  };
  /**
   * Define if table has borders
   */
  bordered?: boolean;
  /**
   * Toggles table header visibility
   */
  showHeader?: boolean;
  /**
   * Custom table styles
   */
  style?: CSSProperties;
}

export interface AssetEventsPanelStyles {
  table?: CSSProperties;
  tableRow?: CSSProperties;
  tableCell?: CSSProperties;
}

export interface MetaEventsProps extends TableDesignType {
  /**
   * Array of objects that customize titles of the columns in the table
   */
  columns?: TableColumnType[];
}

export interface AssetEventsPanelStylesProps {
  /**
   * Object that defines inline CSS styles for inner elements of the component
   */
  styles?: AssetEventsPanelStyles;
}

export interface AssetEventsPanelThemeProps {
  /**
   * @ignore
   */
  theme?: AnyIfEmpty<{}>;
}

export type AssetEventsPanelProps = WithAssetEventsProps &
  MetaEventsProps &
  AssetEventsPanelStylesProps &
  AssetEventsPanelThemeProps;

export type AssetEventsPanelPropsPure = WithAssetEventsDataProps &
  MetaEventsProps &
  AssetEventsPanelStylesProps &
  AssetEventsPanelThemeProps;
