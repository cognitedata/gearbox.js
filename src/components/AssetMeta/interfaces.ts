import { CSSProperties, ReactNode } from 'react';
import { AnyIfEmpty } from '../../interfaces';
import {
  AssetDocumentsPanelStyles,
  MetaDocProps,
} from '../AssetDocumentsPanel';
import { AssetEventsPanelStyles, MetaEventsProps } from '../AssetEventsPanel';
import {
  AssetTimeseriesPanelStyles,
  MetaTimeseriesProps,
} from '../AssetTimeseriesPanel';
import { MetaDescriptionListProps } from '../DescriptionList';

export type AssetPanelType = 'details' | 'events' | 'documents' | 'timeseries';

export interface AssetMetaStyles {
  header?: CSSProperties;
  emptyTab?: CSSProperties;
  details?: CSSProperties;
  timeseries?: AssetTimeseriesPanelStyles;
  documents?: AssetDocumentsPanelStyles;
  events?: AssetEventsPanelStyles;
}

export interface AssetMetaProps {
  /**
   * Asset Id
   */
  assetId: number;
  /**
   * List of panes to be hidden
   */
  hidePanels?: AssetPanelType[];
  /**
   * Defines pane that will be activated once the data has been loaded
   */
  tab?: string;
  /**
   * Function triggered when a user changes panes
   */
  onPaneChange?: (key: string) => void;
  /**
   * Object passed as props to inner component that presents details pane
   */
  detailsProps?: MetaDescriptionListProps;
  /**
   * Object passed as props to inner component that presents timeseries pane
   */
  timeseriesProps?: MetaTimeseriesProps;
  /**
   * Object passed as props to inner component that presents documents pane
   */
  docsProps?: MetaDocProps;
  /**
   * Object passed as props to inner component that presents events pane
   */
  eventProps?: MetaEventsProps;
  /**
   * A custom spinner to be shown in tabs while data is being loaded
   */
  customSpinner?: ReactNode;
  /**
   * Object that defines inline CSS styles for inner elements of the component.
   */
  styles?: AssetMetaStyles;
  /**
   * @ignore
   */
  theme?: AnyIfEmpty<{}>;
}
