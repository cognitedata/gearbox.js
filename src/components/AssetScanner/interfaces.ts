import { Asset } from '@cognite/sdk';
import { ComponentType, CSSProperties, ReactNode } from 'react';
import { Callback, CropSize, EmptyCallback } from '../../interfaces';
import { defaultStrings } from './WebcamScanner/WebcamScanner';

export type AssetScannerStrings = {
  [name in keyof typeof defaultStrings]: string;
};
export type OnAssetListCallback = (assets: Asset[]) => void;
export type SetVideoRefCallback = (element: HTMLVideoElement | null) => void;
export type ASNotification = (type: ASNotifyTypes) => any;
export type ButtonRenderProp = (
  onCapture: Callback,
  isReady: boolean
) => ReactNode;

export interface OcrRequest {
  image: string;
  key?: string;
  extractOcrStrings?: (data: any) => string[];
}

export enum ASNotifyTypes {
  recognizeSuccess = 'recognizeSuccess',
  recognizeFails = 'recognizeFails',
  assetsFind = 'assetsFind',
  assetsEmpty = 'assetsEmpty',
  errorVideoAccess = 'errorVideoAccess',
  errorOccurred = 'errorOccurred',
}

export interface AssetScannerStyles {
  button: CSSProperties;
}

export interface AssetScannerProps {
  /**
   * Function that provides custom OCR call to detect strings on image
   */
  ocrRequest: (ocrParams: OcrRequest) => Promise<string[]>;
  /**
   * Base64 image representation which uses as a source for recognition
   * (instead of webcamera image)
   */
  image?: string;
  /**
   * ocrKey	API key property which is needed if you use embedded Google Vision API
   */
  ocrKey?: string;
  /**
   * Render prop function, that should returns button node. Capture function and
   * image base64 string are passed as arguments
   */
  button?: ButtonRenderProp;
  /**
   * Function that gets a result from the recognize function and formats it as an
   * array of recognized strings (Can be used if you use embedded Google Vision API)
   */
  extractOcrStrings?: (ocrResult: any) => string[];
  /**
   * enableNotification	Flag that controls enabling/disabling notification feature.
   * If customNotification doesn't set then antd messages will be used by default
   */
  enableNotification?: boolean;
  /**
   * Callback function to react on provided type of notification (will be ignored if enableNotification = false)
   */
  customNotification?: ASNotification;
  /**
   * Callback triggered when image recognition process starts
   */
  onImageRecognizeStart?: (image: string) => void;
  /**
   * Callback triggered when image recognition process is finished and right before fetching assets by recognized strings
   */
  onImageRecognizeFinish?: (strings: string[] | null) => void;
  /**
   * Callback triggered if strings haven't been recognized on provided image
   */
  onImageRecognizeEmpty?: Callback;
  /**
   * Callback triggered when SDK asset search has been finished
   */
  onAssetFetchResult?: OnAssetListCallback;
  /**
   * Callback triggered right after taking a shot from camera
   */
  onStartLoading?: EmptyCallback;
  /**
   * Callback triggered after finishing recognition process
   */
  onEndLoading?: EmptyCallback;
  /**
   * Callback triggered when occurs an error related to OCR service/request
   */
  onOcrError?: Callback;
  /**
   * Callback triggered when an error occurs
   */
  onError?: Callback;
  /**
   * Callback triggered after resetting captured image
   */
  onImageReset?: Callback;
  /**
   * styles	Object that defines inline CSS styles for inner elements of the component
   * (Use if you not provide button render prop)
   */
  styles?: AssetScannerStyles;
  /**
   * Object that defines strings to be passed to component
   */
  imageAltText?: string;
  /**
   * Object with strings that can be passed to replace default strings
   */
  strings?: AssetScannerStrings;
  /**
   * Object to specify dimension of crop area
   */
  cropSize?: CropSize;
  /**
   * Component overlaid cropped area
   */
  webcamCropOverlay?: ComponentType;
  /**
   * Function replaces default asset fetching behaviour
   */
  getAssetsHandlerCustom?: (strings: string[]) => Promise<Asset[]>;
}
