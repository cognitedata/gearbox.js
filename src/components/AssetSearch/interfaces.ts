import { Asset } from '@cognite/sdk';
import { Callback, PureObject } from '../../interfaces';
import { SearchStyles } from '../common/Search/interfaces';

export type AssetSearchStyles = SearchStyles;

export interface AssetSearchProps {
  /**
   * Enable root advanced search
   */
  advancedSearch: boolean;
  /**
   * Enable root asset selection
   */
  rootAssetSelect: boolean;
  /**
   * Triggers after selecting one of items from live search results list.
   * Required when showLiveSearchResults == true
   */
  onLiveSearchSelect?: (asset: Asset) => void;
  /**
   * flag to show live search results in dropdown list
   */
  showLiveSearchResults?: boolean;
  /**
   * Triggers when search error occurs
   */
  onError?: Callback;
  /**
   * Object of strings to be placed in component
   */
  strings?: PureObject;
  /**
   * Custom styles
   */
  styles?: AssetSearchStyles;
  /**
   * Triggers when search request finishes
   */
  onSearchResult?: (assets: Asset[]) => void;
}
