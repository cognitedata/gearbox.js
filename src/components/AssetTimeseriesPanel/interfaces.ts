import React from 'react';
import { WithAssetTimeseriesProps } from '../../hoc';
import { AnyIfEmpty } from '../../interfaces';
import { TimeseriesChartMetaProps } from '../TimeseriesChartMeta';

export interface AssetTimeseriesPanelStyles {
  wrapper?: React.CSSProperties;
  timeseriesContainer?: React.CSSProperties;
}

export interface MetaTimeseriesProps
  extends Omit<TimeseriesChartMetaProps, 'timeseriesId'> {
  /**
   * Strings that can be customized
   */
  strings?: {
    noTimeseriesSign?: string;
  };
  /**
   * Custom styles for component
   */
  styles?: AssetTimeseriesPanelStyles;
  /**
   * @ignore
   */
  theme?: AnyIfEmpty<{}>;
}

export type AssetTimeseriesPanelProps = WithAssetTimeseriesProps &
  MetaTimeseriesProps;
