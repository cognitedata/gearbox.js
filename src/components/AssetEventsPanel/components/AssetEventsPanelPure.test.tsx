import { CogniteEvent } from '@cognite/sdk';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { ASSET_META_EVENTS_STYLES, fakeEvents } from '../../../mocks';
import { EventAddonsProp, MetadataRenderer } from '../interfaces';
import {
  AssetEventsPanelPure,
  AssetEventsPanelPureComponent,
} from './AssetEventsPanelPure';

configure({ adapter: new Adapter() });

describe('AssetEventsPanel', () => {
  it('checks if the modal is opened', done => {
    const eventPanelModal = mount(
      <AssetEventsPanelPure assetEvents={fakeEvents} />
    );

    expect(
      eventPanelModal.find(AssetEventsPanelPureComponent).state('selectedEvent')
    ).toEqual(null);

    const clickableRow = eventPanelModal.find('td').first();
    clickableRow.simulate('click');
    setImmediate(() => {
      eventPanelModal.update();
      const modalWindow = eventPanelModal.find('Modal');
      expect(modalWindow).toHaveLength(1);
      done();
    });
  });

  it('should allow custom rendering of metadata', done => {
    const assetEvents = fakeEvents;

    const eventCssId = (event: CogniteEvent) => `some-${event.id}`;

    const eventMessage = (event: CogniteEvent) => {
      const source = (event.metadata && event.metadata.source) || '';
      return `${source}: lorem-ipsum`;
    };

    const renderEventMetadata: MetadataRenderer = (event: EventAddonsProp) => (
      <div key={event.id} id={eventCssId(event)}>
        {eventMessage(event)}
      </div>
    );

    const eventPanelModal = mount(
      <AssetEventsPanelPure {...{ assetEvents, renderEventMetadata }} />
    );

    const clickableRow = eventPanelModal.find('td').first();
    clickableRow.simulate('click');

    setImmediate(() => {
      eventPanelModal.update();

      const [firstEvent] = fakeEvents;
      const eventMetadata = eventPanelModal.find(`#${eventCssId(firstEvent)}`);

      const expectedMessage = eventMessage(firstEvent);
      expect(eventMetadata.text()).toEqual(expectedMessage);
      done();
    });
  });

  it.each`
    inlineStyle                           | element
    ${ASSET_META_EVENTS_STYLES.table}     | ${'.ant-table-wrapper'}
    ${ASSET_META_EVENTS_STYLES.tableRow}  | ${'tr'}
    ${ASSET_META_EVENTS_STYLES.tableCell} | ${'td'}
  `(
    'should render component with passed inline styles',
    ({ inlineStyle, element }) => {
      const eventPanel = mount(
        <AssetEventsPanelPure
          assetEvents={fakeEvents}
          styles={ASSET_META_EVENTS_STYLES}
        />
      );
      const containerStyle = eventPanel
        .find(element)
        .last()
        .prop('style');

      expect(inlineStyle === containerStyle).toBeTruthy();
    }
  );
});
