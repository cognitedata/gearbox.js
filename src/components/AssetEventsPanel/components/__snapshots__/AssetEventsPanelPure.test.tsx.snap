// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetEventsPanel renders correctly 1`] = `
<div
  className="ant-table-wrapper sc-bdVaJa bDWFJH"
>
  <div
    className="ant-spin-nested-loading"
  >
    <div
      className="ant-spin-container"
    >
      <div
        className="ant-table ant-table-default ant-table-scroll-position-left"
        style={Object {}}
      >
        <div
          className="ant-table-content"
        >
          <div
            className="ant-table-body"
            onScroll={[Function]}
            onWheel={[Function]}
            style={Object {}}
          >
            <table
              className=""
              style={Object {}}
            >
              <colgroup>
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
              </colgroup>
              <thead
                className="ant-table-thead"
              >
                <tr
                  style={
                    Object {
                      "height": null,
                    }
                  }
                >
                  <th
                    className=""
                  >
                    <span
                      className="ant-table-header-column"
                    >
                      <div>
                        <span
                          className="ant-table-column-title"
                        >
                          Type / Subtype
                        </span>
                        <span
                          className="ant-table-column-sorter"
                        />
                      </div>
                    </span>
                  </th>
                  <th
                    className=""
                  >
                    <span
                      className="ant-table-header-column"
                    >
                      <div>
                        <span
                          className="ant-table-column-title"
                        >
                          Description
                        </span>
                        <span
                          className="ant-table-column-sorter"
                        />
                      </div>
                    </span>
                  </th>
                  <th
                    className=""
                  >
                    <span
                      className="ant-table-header-column"
                    >
                      <div>
                        <span
                          className="ant-table-column-title"
                        >
                          Start
                        </span>
                        <span
                          className="ant-table-column-sorter"
                        />
                      </div>
                    </span>
                  </th>
                  <th
                    className=""
                  >
                    <span
                      className="ant-table-header-column"
                    >
                      <div>
                        <span
                          className="ant-table-column-title"
                        >
                          End
                        </span>
                        <span
                          className="ant-table-column-sorter"
                        />
                      </div>
                    </span>
                  </th>
                </tr>
              </thead>
              <tbody
                className="ant-table-tbody"
              >
                <tr
                  className="sc-bwzfXH kByjcP"
                  onClick={[Function]}
                >
                  <td
                    className="sc-htpNat irDPrN"
                  >
                    <span
                      className="ant-table-row-indent indent-level-0"
                      style={
                        Object {
                          "paddingLeft": "0px",
                        }
                      }
                    />
                    <span>
                      // @ts-ignore // TODO: Remove @ts-ignore one SDK will be fixed
                      <strong
                        style={
                          Object {
                            "display": "block",
                          }
                        }
                      />
                      // @ts-ignore // TODO: Remove @ts-ignore one SDK will be fixed
                      <span
                        style={
                          Object {
                            "fontSize": 12,
                            "opacity": 0.8,
                          }
                        }
                      />
                    </span>
                  </td>
                  <td
                    className="sc-htpNat irDPrN"
                  >
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                  </td>
                  <td
                    className="sc-htpNat irDPrN"
                  >
                    April 24, 2019 3:35 PM
                  </td>
                  <td
                    className="sc-htpNat irDPrN"
                  >
                    Ongoing
                  </td>
                </tr>
                <tr
                  className="sc-bwzfXH kByjcP"
                  onClick={[Function]}
                >
                  <td
                    className="sc-htpNat irDPrN"
                  >
                    <span
                      className="ant-table-row-indent indent-level-0"
                      style={
                        Object {
                          "paddingLeft": "0px",
                        }
                      }
                    />
                    <span>
                      // @ts-ignore // TODO: Remove @ts-ignore one SDK will be fixed
                      <strong
                        style={
                          Object {
                            "display": "block",
                          }
                        }
                      />
                      // @ts-ignore // TODO: Remove @ts-ignore one SDK will be fixed
                      <span
                        style={
                          Object {
                            "fontSize": 12,
                            "opacity": 0.8,
                          }
                        }
                      />
                    </span>
                  </td>
                  <td
                    className="sc-htpNat irDPrN"
                  >
                    sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
                  </td>
                  <td
                    className="sc-htpNat irDPrN"
                  >
                    April 27, 2019 3:35 PM
                  </td>
                  <td
                    className="sc-htpNat irDPrN"
                  >
                    April 30, 2019 3:35 PM
                  </td>
                </tr>
                <tr
                  className="sc-bwzfXH kByjcP"
                  onClick={[Function]}
                >
                  <td
                    className="sc-htpNat irDPrN"
                  >
                    <span
                      className="ant-table-row-indent indent-level-0"
                      style={
                        Object {
                          "paddingLeft": "0px",
                        }
                      }
                    />
                    <span>
                      // @ts-ignore // TODO: Remove @ts-ignore one SDK will be fixed
                      <strong
                        style={
                          Object {
                            "display": "block",
                          }
                        }
                      />
                      // @ts-ignore // TODO: Remove @ts-ignore one SDK will be fixed
                      <span
                        style={
                          Object {
                            "fontSize": 12,
                            "opacity": 0.8,
                          }
                        }
                      />
                    </span>
                  </td>
                  <td
                    className="sc-htpNat irDPrN"
                  >
                    Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
                  </td>
                  <td
                    className="sc-htpNat irDPrN"
                  >
                    May 7, 2019 3:35 PM
                  </td>
                  <td
                    className="sc-htpNat irDPrN"
                  >
                    May 14, 2019 3:35 PM
                  </td>
                </tr>
                <tr
                  className="sc-bwzfXH kByjcP"
                  onClick={[Function]}
                >
                  <td
                    className="sc-htpNat irDPrN"
                  >
                    <span
                      className="ant-table-row-indent indent-level-0"
                      style={
                        Object {
                          "paddingLeft": "0px",
                        }
                      }
                    />
                    <span>
                      // @ts-ignore // TODO: Remove @ts-ignore one SDK will be fixed
                      <strong
                        style={
                          Object {
                            "display": "block",
                          }
                        }
                      />
                      // @ts-ignore // TODO: Remove @ts-ignore one SDK will be fixed
                      <span
                        style={
                          Object {
                            "fontSize": 12,
                            "opacity": 0.8,
                          }
                        }
                      />
                    </span>
                  </td>
                  <td
                    className="sc-htpNat irDPrN"
                  >
                    Excepteur sint occaecat cupidatat non proident
                  </td>
                  <td
                    className="sc-htpNat irDPrN"
                  >
                    April 4, 2019 3:35 PM
                  </td>
                  <td
                    className="sc-htpNat irDPrN"
                  >
                    April 14, 2019 3:35 PM
                  </td>
                </tr>
                <tr
                  className="sc-bwzfXH kByjcP"
                  onClick={[Function]}
                >
                  <td
                    className="sc-htpNat irDPrN"
                  >
                    <span
                      className="ant-table-row-indent indent-level-0"
                      style={
                        Object {
                          "paddingLeft": "0px",
                        }
                      }
                    />
                    <span>
                      // @ts-ignore // TODO: Remove @ts-ignore one SDK will be fixed
                      <strong
                        style={
                          Object {
                            "display": "block",
                          }
                        }
                      />
                      // @ts-ignore // TODO: Remove @ts-ignore one SDK will be fixed
                      <span
                        style={
                          Object {
                            "fontSize": 12,
                            "opacity": 0.8,
                          }
                        }
                      />
                    </span>
                  </td>
                  <td
                    className="sc-htpNat irDPrN"
                  >
                    Sunt in culpa qui officia deserunt mollit anim id est laborum.
                  </td>
                  <td
                    className="sc-htpNat irDPrN"
                  >
                    June 6, 2019 3:35 PM
                  </td>
                  <td
                    className="sc-htpNat irDPrN"
                  >
                    June 13, 2019 3:35 PM
                  </td>
                </tr>
                <tr
                  className="sc-bwzfXH kByjcP"
                  onClick={[Function]}
                >
                  <td
                    className="sc-htpNat irDPrN"
                  >
                    <span
                      className="ant-table-row-indent indent-level-0"
                      style={
                        Object {
                          "paddingLeft": "0px",
                        }
                      }
                    />
                    <span>
                      // @ts-ignore // TODO: Remove @ts-ignore one SDK will be fixed
                      <strong
                        style={
                          Object {
                            "display": "block",
                          }
                        }
                      />
                      // @ts-ignore // TODO: Remove @ts-ignore one SDK will be fixed
                      <span
                        style={
                          Object {
                            "fontSize": 12,
                            "opacity": 0.8,
                          }
                        }
                      />
                    </span>
                  </td>
                  <td
                    className="sc-htpNat irDPrN"
                  >
                    No description
                  </td>
                  <td
                    className="sc-htpNat irDPrN"
                  >
                    -
                  </td>
                  <td
                    className="sc-htpNat irDPrN"
                  >
                    Ongoing
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <ul
        className="ant-pagination ant-table-pagination"
        style={Object {}}
        unselectable="unselectable"
      >
        <li
          aria-disabled={true}
          className="ant-pagination-disabled ant-pagination-prev"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={null}
          title="Previous Page"
        >
          <a
            className="ant-pagination-item-link"
          >
            <i
              aria-label="icon: left"
              className="anticon anticon-left"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="left"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                />
              </svg>
            </i>
          </a>
        </li>
        <li
          className="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title={1}
        >
          <a>
            1
          </a>
        </li>
        <li
          aria-disabled={true}
          className="ant-pagination-disabled ant-pagination-next"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={null}
          title="Next Page"
        >
          <a
            className="ant-pagination-item-link"
          >
            <i
              aria-label="icon: right"
              className="anticon anticon-right"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="right"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
                />
              </svg>
            </i>
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
